src/parser/mod.rs

```diff
@@ -6338,6 +6345,15 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub fn parse_array_join_table_factor(&mut self) -> Result<TableFactor, ParserError> {
+        {
+            let expr = self.parse_expr()?;
+
+            let alias = self.parse_optional_table_alias(keywords::RESERVED_FOR_TABLE_ALIAS)?;
```

Comment:
I am not sure about ARRAY JOIN always being a "field accessor" -- would't it makes sense to parse the argument as [TableFactor](https://docs.rs/sqlparser/latest/sqlparser/ast/enum.TableFactor.html#) and that way it can support being a table / subquery / etc rather than just an expression? That would also handle aliases as well

Reply:
I don't fully understand what TableFactor is or its purpose in the lib 🤔 I will have to give it more read.
I checked Clickhouse's latest docs and there is more syntax I need to support e.g.

```sql
SELECT s, `nest.x`, `nest.y`
FROM nested_test
ARRAY JOIN `nest.x`, `nest.y`;
```

From what I could tell, this behaves much like UNNEST, and there are important limitations to what could be inside such `ARRAY JOIN` hence I'm not so sure if making it generic TableFactor actually makes sense 🤔

Reply:
I think of `TableFactor` as basically anything that could be JOINEd (aka show up as a element in the FROM clause)